### ISSUE TITLE
Regression in self-hosted gitlab releases after switching to native fetch

### ISSUE THREAD

**Issue by vonBrax** [release-it/release-it#1184]
Hi again!

A while ago I submitted a PR #1129  to fix #1125 to allow insecure connections to self hosted gitlab instances.
The problem is back again after removing `node-fetch` and `got` in favor of native `fetch` on `v17.7.0` .
Additionally, the `certificateAuthorityFile` option also no longer works.

To make these settings work with the native `fetch`, we need to pass down an `Agent` instance from [undici](https://github.com/nodejs/undici) to the `fetch` ([see comment](https://github.com/nodejs/node/issues/48977#issuecomment-1658324420)).

`Node` uses `undici` internally for the fetch requests and currently does not publicly expose it, but there is an [open issue](https://github.com/nodejs/node/issues/43187) to expose some of its APIs.

I could submit a PR to update the `fetch` configuration to make these settings work again, but that would mean that we would need to add `undici` as a new dependency, at least temporarily until `node` exposes the `Agent` class.

If that would be an acceptable solution, let me know and I'll create the PR.

I'm also open to suggestions if there are any other better solutions.

**Comment by webpro** [comment]
Sorry for the hassle, it was a bit unfortunate without having tests for the case. A PR for this would be fine with me.

**Comment by vonBrax** [comment]
Yeah, it would be nice if we could actually test these settings, but I'm not sure if that's something that could easily be done. I'll see if I can come up with something, maybe spinning up a local server or something.
I'll prepare the PR.
Thank you for your prompt answer!

**Comment by webpro** [comment]
The issue seems to have been addressed, closing this one.
